examples/vqe_toric.py

```diff
@@ -0,0 +1,609 @@
+"""
+Variational Quantum Eigensolver (VQE) for the Toric Code Model
+
+This module implements VQE optimization for finding ground states of the
+generalized 2D toric code Hamiltonian with open boundary conditions and provides comparison
+among different ansätze: FLDC, GLDC, and FDC.
+
+"""
+
+import numpy as np
+import tensorflow as tf
+import scipy.sparse as sp
+from scipy.sparse.linalg import eigs
+from itertools import product
+from typing import Dict, Tuple, Optional
+import matplotlib.pyplot as plt
+
+import tensorcircuit as tc
+from tensorcircuit.quantum import PauliStringSum2COO_tf
```

Comment:
better use PauliStringSum2COO, the backend can automatically follow tc.set_backend()